hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/handler/AssignRegionHandler.java

```diff
@@ -105,50 +102,54 @@ public void process() throws IOException {
     if (previous != null) {
```

Comment:
nit: not your code, but i liked better the three-way if-elseif-else structure from `HRegionServer#closeRegion`. Makes it nice and obvious that there are three states possible from the return value. It would also be nice to have a log message in all three cases, rather than silently doing nothing for the `previous == null` case.